mitodl/mitxpro
# Checkout page

#### Pre-Flight checklist


#### What are the relevant tickets?
Fixes #76 

#### What's this PR do?
Implements a checkout page

#### How should this be manually tested?
 - Get the test credentials for CyberSource from @noisecapella 
 - Set up your basket and a coupon:

        username = "..."
        user = User.objects.get(username=username)
        basket, _ = Basket.objects.get_or_create(user=user)
        course = Course.objects.all()[2]
        product = Product.objects.create(content_object=course)
        product_version = ProductVersion.objects.create(product=product, price=123.45, description="A product for a course")
        BasketItem.objects.filter(basket=basket).delete()
        BasketItem.objects.create(product=product, basket=basket, quantity=1)
        
        coupon_payment = CouponPayment.objects.create(name='pr108')
        coupon_payment_version = CouponPaymentVersion.objects.create(tag='pr108', payment=coupon_payment, coupon_type="credit_card", num_coupon_codes=1, max_redemptions=1, max_redemptions_per_user=1, amount=0.5)
        coupon = Coupon.objects.create(payment=coupon_payment, coupon_code='pr108', enabled=True)
        CouponVersion.objects.create(coupon=coupon, payment_version=coupon_payment_version)
        
        CouponEligibility.objects.create(coupon=coupon, product=product)

 - Go to `/checkout/` and verify that you can click 'checkout' and are redirected to the checkout page. 
 - Go through cybersource. Credit card number is 4111111111111111, any expiration number and three digit CVV should work. 

Order fulfillment: I will receive an email for the attempt to purchase, I'll forward it to you. Then from a Python shell type `requests.post("/api/order_fulfillment/", payload)` where `payload` is the dict created from the key value pairs in the email. You should get a 200 response and the `Order` created should be marked `fulfilled`.

#### Screenshots (if appropriate)
Without coupon:
![without_coupon](https://user-images.githubusercontent.com/863262/55972225-1cfd9d00-5c51-11e9-9ab9-e38f1ac4cd82.png)

With coupon:
![with_coupon](https://user-images.githubusercontent.com/863262/55972245-2555d800-5c51-11e9-9fad-0c7eef71ad0c.png)
